projectmanagement/src/main/java/life/qbic/projectmanagement/application/ProjectCreationService.java

```diff
@@ -0,0 +1,54 @@
+package life.qbic.projectmanagement.application;
+
+import life.qbic.application.commons.ApplicationException;
+import life.qbic.application.commons.ApplicationException.ErrorCode;
+import life.qbic.application.commons.ApplicationException.ErrorParameters;
+import life.qbic.application.commons.Result;
+import life.qbic.projectmanagement.domain.Project;
+import life.qbic.projectmanagement.domain.ProjectIntent;
+import life.qbic.projectmanagement.domain.ProjectRepository;
+import life.qbic.projectmanagement.domain.ProjectTitle;
+
+/**
+ * Application service facilitating the creation of projects.
+ */
+public class ProjectCreationService {
+
+  private final ProjectRepository projectRepository;
+
+  public ProjectCreationService(ProjectRepository projectRepository) {
+    this.projectRepository = projectRepository;
+  }
+
+  /**
+   * Create a new project based on the information provided.
+   *
+   * @param title the title of the project.
+   * @return the created project
+   */
+  public Result<Project, ApplicationException> createProject(String title) {
+    ProjectTitle projectTitle;
+    try {
+      projectTitle = new ProjectTitle(title);
+    } catch (Exception e) {
```

Comment:
Which exception is thrown here? We should avoid catching generic exceptions, as it prohibits us from distinguishing what happened

Reply:
This does not matter as it is not swallowed but wrapped.

Reply:
We want to wrap all exceptions from the domain layer

Reply:
do we? this way we expose domain details to the client via the application service, which we dont want. That is the hole point of introducing the error code.